scripts/CatapultRecipeUpdater.py

```diff
@@ -0,0 +1,191 @@
+		recipe_with_updates = dict(ChainMap(*results))
+		return recipe_with_updates
+
+	async def _update_conandata_sha(self, recipe, filepath, version):
+		config = await self._read_yaml_file(filepath)
+		url = config['sources'][version]['url']
+		with tempfile.TemporaryDirectory() as tmpdir:
+			download_file = f'{tmpdir}/{recipe}.{version}.tar.gz'
+			dispatch_subprocess(['curl', '-LJ', url, '--output', f'{download_file}'])
+			output, _ = dispatch_subprocess(['sha256sum', f'{download_file}'], handle_error=True)
+			config['sources'][version]['sha256'] = output.split(' ')[0]
+			with open(filepath, 'w') as output_file:
+				yaml.dump(config, output_file, sort_keys=False)
+
+	async def _update_recipe_files(self, recipe, versions, recipes_versions):
+		update_recipe_files_descriptor = [
+			r'{recipe}/config.yml',
+			r'{recipe}/all/conandata.yml',
+			r'{recipe}/all/test_package/CMakeLists.txt',
+			r'{recipe}/all/conanfile.py',
+		]
+		dependency_map = {'mongo-cxx-driver': 'mongo-c-driver', 'cppzmq': 'zeromq'}
+
+		print(f'updating recipe {recipe} {versions}')
+		for recipe_file_descriptor in update_recipe_files_descriptor:
+			recipe_file = recipe_file_descriptor.format(recipe=recipe)
+			recipe_file_path = self.source_path / recipe_file
+			if recipe_file_path.exists():
+				if 'conanfile.py' == recipe_file_path.name:
+					if recipe not in dependency_map.keys():
+						continue
+
+					dependency_recipe = dependency_map[recipe]
+					if dependency_recipe not in recipes_versions.keys():
+						continue
+
+					current_version, new_version = recipes_versions[dependency_map[recipe]]
+				else:
+					current_version, new_version = versions
+
+				dispatch_subprocess(
+					['sed', '-i', f's/{current_version}/{new_version}/g', str(recipe_file_path)],
+					handle_error=True
+				)
+
+				if 'conandata.yml' == recipe_file_path.name:
+					await self._update_conandata_sha(recipe, recipe_file_path, new_version)
+
+	async def update_recipe_version(self, recipes_versions):
+		print(f'updating recipes {recipes_versions}')
+		await asyncio.gather(*[
+			self._update_recipe_files(recipe, versions, recipes_versions) for recipe, versions in recipes_versions.items()
+		])
+
+	async def build_conan_package(self, recipes_versions):
+		await self._initialize_conan()
+		for recipe, versions in recipes_versions.items():
+			_, new_version = versions
+			recipe_path = self.source_path / f'{recipe}/all'
+			dispatch_subprocess(
+				['conan', 'create', '.', f'{new_version}@nemtech/stable', '--remote=nemtech'],
+				cwd=recipe_path,
+				handle_error=True
+			)
+
+	async def upload_conan_package(self, recipes_versions):
+		for recipe, versions in recipes_versions:
+			_, new_version = versions
+			recipe_path = self.source_path / f'{recipe}/all'
+			dispatch_subprocess(
+				['conan', 'upload', f'{recipe}/{new_version}@nemtech/stable', '--remote=nemtech'],
+				cwd=recipe_path,
+				handle_error=True
+			)
```

Comment:
consider refactoring as this is same as previous with only difference being conan command